exasol_integration_test_docker_environment/lib/test_environment/ports.py

```diff
@@ -0,0 +1,40 @@
+import socket
+from contextlib import ExitStack
+from typing import List, Optional
+
+
+def find_free_ports(num_ports: int) -> List[int]:
+    def new_socket():
+        return socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    def bind(sock: socket.socket, port: int):
+        sock.bind(('', port))
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+
+    result = list()
+    with ExitStack() as stack:
+        sockets = [stack.enter_context(new_socket()) for dummy in range(num_ports)]
+        for sock in sockets:
+            bind(sock, 0)
+            result.append(sock.getsockname()[1])
+    with ExitStack() as stack:
+        sockets_and_ports = [(stack.enter_context(new_socket()), port) for port in result]
+        for sock, port in sockets_and_ports:
+            bind(sock, port)
+    return result
+
+
+class PortForwarding:
+    def __init__(self, database: int, bucketfs: int, ssh: Optional[int] = None):
+        self.database = database
+        self.bucketfs = bucketfs
+        self.ssh = ssh
+
+    @classmethod
+    def random_free(cls, ssh: bool = True):
+        ports = find_free_ports(3 if ssh else 2) + [None]
+        return PortForwarding(*ports[:3])
+
+    @classmethod
+    @property
+    def default_ports(cls) -> 'PortForwarding':
+        return PortForwarding(database=8888, bucketfs=6583, ssh=22)
```

Comment:
use maybe constants for the ports, such that we use everywhere the same defaults

Reply:
My idea was to understand attribute `PortForwarding.default_ports` as a "constant".
I see the benefit of having a clear, short, yet precise naming, with class name as context and all information in a single place.

So if any of the lib/api/testing/etc. would need to reference one of the default ports, it could just write `PortForwarding.default_ports.ssh` for example, which is very literate and "speaking" from my perspective.